### PR TITLE
Make the bootstrap command perform a fake bootstrap

### PR DESCRIPTION
**Pull request by freeekanayaka**
The fake-juju cli bootstrap command now posts an HTTP request against
fake-jujud's control plane API in order to perform a fake "bootstrap",
where fake-jujud starts a new controller machine (machine 0).

The logic to setup a custom test certificate has been moved to the
juju core overlay source tree since it's now used by the command
line client too.

The full end-to-end process is exercised by new python integration
test assertions.

### CODE REVIEW

File: common/core/github.com/juju/juju/cmd/juju/commands/bootstrap-fakejuju.go [hunk 1]
```diff
@@ -28,6 +27,11 @@ func (c *bootstrapCommand) fakeJujuBootstrap() error {
 	controller := c.controllerName
 	model := "controller"
 
+	logger.Debugf("bootstrapping %s:%s", controller, model)
+
+	if err := testing.SetCerts(os.Getenv("FAKE_JUJUD_CERTS")); err != nil {
+	}
```
Comment (bjornt) [inline common/core/github.com/juju/juju/cmd/juju/commands/bootstrap-fakejuju.go]: Shouldn't you return err here?

File: python/tests/test_fixture.py [hunk 2]
```diff
@@ -61,28 +68,20 @@ def test_up(self):
...
         self.useFixture(EnvironmentVariable("JUJU_DATA", juju_data.path))
+        self.useFixture(EnvironmentVariable("FAKE_JUJUD_CERTS", CERT))
         check_call([FAKE_JUJU, "bootstrap", "foo", "bar"])
+        check_call([FAKE_JUJU, "switch", "bar"], stdout=PIPE, stderr=STDOUT)
+
+        output = check_output([FAKE_JUJU, "status", "--format=json"])
+        status = json.loads(output)
+
+        self.assertEqual(
+            "pending", status["machines"]["0"]["machine-status"]["current"])
```
Comment (bjornt) [inline python/tests/test_fixture.py]: In the PR description and the test docstring you say that the machine is "started". Why do you ensure that it's "pending" here?

File: python/tests/test_fixture.py [hunk 3]
```diff
@@ -61,28 +68,20 @@ def test_up(self):
...
-        with open(juju_data.join("models.yaml")) as fd:
-            models = yaml.load(fd)
-            self.assertEqual(
-                {"admin/controller": {"uuid": MODEL_UUID}},
-                models["controllers"]["bar"]["models"])
-            self.assertEqual(
-                "admin/controller",
-                models["controllers"]["bar"]["current-model"])
+        self.assertEqual(
+            "pending", status["machines"]["0"]["juju-status"]["current"])
```
Comment (bjornt) [inline python/tests/test_fixture.py]: Same comment as above.

File: common/core/github.com/juju/juju/cmd/juju/commands/bootstrap-fakejuju.go (hunk 1 above)
Comment (freeekanayaka) [inline common/core/github.com/juju/juju/cmd/juju/commands/bootstrap-fakejuju.go]: Yes I should. Fixed.

File: python/tests/test_fixture.py (hunk 2 above)
Comment (freeekanayaka) [inline python/tests/test_fixture.py]: Ah sorry, initially this branch had the logic to transition to the "started" state, but it had become a bit biggish, so I extracted the change in a separate branch (that will follow up). The PR description comes for a commit message that I amended without modifying it.

File: python/tests/test_fixture.py (hunk 3 above)
Comment (freeekanayaka) [inline python/tests/test_fixture.py]: Same.

**Pull request merged**